libflux/flux-core/src/semantic/nodes.rs

```diff
@@ -867,6 +867,15 @@ impl FunctionExpr {
         }
         // And use it to infer the body.
         let (nenv, bcons) = self.body.infer(nenv, f)?;
+        // HACK Remove once substitutions are persisted correctly
+        // Update the argument variable with the types inferred from the body
+        for (k, t) in req.iter_mut().chain(opt.iter_mut()) {
+            *t = nenv
+                .lookup(k)
+                .expect("The binding should have been added above")
```

Comment:
Does this panic? Unfortunately we cannot panic in this code, since it's invoked from Go, and (iirc) it can cause the server process to exit.

Reply:
It could, of course, but only due to bug elsewhere as the binding that it refers to were defined above in this same function.

Reply:
I think we should still avoid uses of `expect` in this code, and return an error. I know it's a pain, but if we never call code that panics (and I don't think we do in our code, save for 3rd party code) then we don't have to worry about taking down the whole process.